powerauth-data-adapter/src/main/java/io/getlime/security/powerauth/app/dataadapter/impl/service/DataAdapterService.java

```diff
@@ -190,6 +190,46 @@ public void formDataChangedNotification(String userId, String organizationId, Fo
         throw new IllegalStateException("Invalid change entity type: " + change.getType());
     }
 
+    @Override
+    public CreateImplicitLoginOperationResponse createImplicitLoginOperation(String clientId, String[] scopes) throws DataAdapterRemoteException {
+        try {
+            final TppAppDetailResponse appDetail = tppEngineService.fetchAppDetail(clientId);
+
+            // Make sure there is only one item in scopes
+            if (scopes == null || scopes.length != 1) {
+                return null;
+            }
+            // Make sure the scope is from known enum
+            final String scope = scopes[0].toLowerCase();
+            if (!"aisp".equals(scope) && !"pisp".equals(scope)) {
+                return null;
+            }
+
+            // Build application context
+            ApplicationContext appContext = new ApplicationContext();
+            appContext.setId(clientId);
+            appContext.setName(appDetail.getName());
+            appContext.setDescription(appDetail.getDescription());
+            appContext.getOriginalScopes().add(scope);
+
+            // Build form data
+            FormData formData = new FormData();
+            formData.addTitle("login.title");
+            formData.addGreeting("login.greeting");
+            formData.addSummary("login.summary");
+
+            // Create an implicit operation
+            CreateImplicitLoginOperationResponse result = new CreateImplicitLoginOperationResponse();
+            result.setName("login_sca");
+            result.setFormData(formData);
+            result.setApplicationContext(appContext);
+
+            return result;
+        } catch (InvalidAppException e) {
```

Comment:
Where does this exception come from? Perhaps I am missing some code changes in `powerauth-webflow`.

Reply:
... `features/psd2` branch...